### DISCUSSION TITLE
ESP32 Record to WAV Example Code -- No Desired Tools Under Drop Down Menu

### DISCUSSION THREAD

**Discussion by jlapp12** [espressif/arduino-esp32#10014] [Q&A] [answered]
Hello,

I am working on a project using the ESP32 libraries and I want to try out the example code of Record to WAV. (from https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP_I2S/examples/Record_to_WAV/Record_to_WAV.ino) . There are some changes I need to make inside the program before I run it. These changes are in the Tools menu.

As commented inside the .ino file, " Don't forget to select the OPI PSRAM, 8MB flash size and Enable USB CDC on boot in the Tools menu!"

When I go to the drop down menu, there are none of these options available. I have the correct board package, and have updated the board to the latest changes. Is there something I am missing? I am working on a Wrover board.

I have attached what I am seeing below. 

![toolsOutput](https://github.com/espressif/arduino-esp32/assets/148792948/89d43691-737b-4266-94da-be1cc075c55c)

Thank you.

**Comment by me-no-dev** [comment, marked as answer]
For Wrover module, you are already good to go. OPI and USB CDC are for ESP32-S3

**Reply by jlapp12** [reply]
Thank you for your quick response.
